tests/integration/components/o-s-s/side-panel-test.ts

```diff
@@ -0,0 +1,136 @@
+      </OSS::SidePanel>`
+      );
+
+      assert.dom('.oss-side-panel__header').hasText('This is the header');
+    });
+
+    test('The content named-block is properly displayed', async function (assert) {
+      await render(
+        hbs`
+      <OSS::SidePanel @title="Sidepanel" @visible={{true}} @backdrop={{true}} @size="sm" @onClose={{this.onClose}}>
+        <:content>
+          <p>This is the content</p>
+        </:content>
+      </OSS::SidePanel>`
+      );
+
+      assert.dom('.oss-side-panel__content').hasText('This is the content');
+    });
+
+    test('The footer named-block is properly displayed', async function (assert) {
+      await render(
+        hbs`
+      <OSS::SidePanel @title="Sidepanel" @visible={{true}} @backdrop={{true}} @size="sm" @onClose={{this.onClose}}>
+        <:footer>
+          <p>This is the footer</p>
+        </:footer>
+      </OSS::SidePanel>`
+      );
+
+      assert.dom('.oss-side-panel__footer').hasText('This is the footer');
+    });
+  });
+
+  module('The visibility of the side panel varies', function () {
+    test('When @visible is truthy, the side panel is visible', async function (assert) {
+      await render(
+        hbs`<OSS::SidePanel @title="Sidepanel" @visible={{true}} @backdrop={{true}} @onClose={{this.onClose}} />`
+      );
+      assert.dom('.oss-side-panel__container.oss-side-panel__container--visible').hasStyle({
+        opacity: '1',
+        visibility: 'visible'
+      });
+    });
+
+    test('When @visible is falsy, the side panel is not visible', async function (assert) {
+      await render(
+        hbs`<OSS::SidePanel @title="Sidepanel" @visible={{false}} @backdrop={{true}} @onClose={{this.onClose}} />`
+      );
+      assert.dom('.oss-side-panel__container').hasStyle({
+        opacity: '0',
+        visibility: 'hidden'
+      });
+    });
+  });
+
+  module('The width of the side panel varies', function () {
+    test('When @size is sm, the width is small', async function (assert) {
+      await render(
+        hbs`<OSS::SidePanel @title="Sidepanel" @visible={{true}} @backdrop={{true}} @size="sm" @onClose={{this.onClose}} />`
+      );
+      assert.dom('.oss-side-panel').hasStyle({
+        width: '480px'
+      });
+    });
+
+    test('When @size is md, the width is medium', async function (assert) {
+      await render(
+        hbs`<OSS::SidePanel @title="Sidepanel" @visible={{true}} @backdrop={{true}} @size="md" @onClose={{this.onClose}} />`
+      );
+      assert.dom('.oss-side-panel').hasStyle({
+        width: '540px'
+      });
+    });
+
+    test('When @size is not defined, the default width is small', async function (assert) {
+      await render(
+        hbs`<OSS::SidePanel @title="Sidepanel" @visible={{true}} @backdrop={{true}} @onClose={{this.onClose}} />`
+      );
+      assert.dom('.oss-side-panel').hasStyle({
+        width: '480px'
+      });
+    });
+  });
+
+  module('Clicking outside the side panel closes it', function () {
```

Comment:
clicking outside the side panel (or Clicking on the background) triggers the @onClose callback